install-opentelemetry-dotnet.ps1

```diff
@@ -0,0 +1,163 @@
+    Architecture of CLR profiler binaries to be installed.
+    Possible values: <auto>, x64, x86
+#>
+[cmdletbinding()]
+param(
+    [string]$OS="windows",
+    [string]$Architecture="<auto>"
+)
+
+function Get-Machine-Architecture() {
+    # On PS x86, PROCESSOR_ARCHITECTURE reports x86 even on x64 systems.
+    # To get the correct architecture, we need to use PROCESSOR_ARCHITEW6432.
+    # PS x64 doesn't define this, so we fall back to PROCESSOR_ARCHITECTURE.
+    # Possible values: amd64, x64, x86, arm64, arm
+    if( $ENV:PROCESSOR_ARCHITEW6432 -ne $null ) {
+        return $ENV:PROCESSOR_ARCHITEW6432
+    }
+
+    try {        
+        if( ((Get-CimInstance -ClassName CIM_OperatingSystem).OSArchitecture) -like "ARM*") {
+            if( [Environment]::Is64BitOperatingSystem )
+            {
+                return "arm64"
+            }  
+            return "arm"
+        }
+    }
+    catch {
+        # Machine doesn't support Get-CimInstance
+    }
+
+    return $ENV:PROCESSOR_ARCHITECTURE
+}
+
+function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
+    if ($Architecture -eq "<auto>") {
+        $Architecture = Get-Machine-Architecture
+    }
+
+    switch ($Architecture.ToLowerInvariant()) {
+        { ($_ -eq "amd64") -or ($_ -eq "x64") } { return "x64" }
+        { $_ -eq "x86" } { return "x86" }
+        default { throw "Architecture '$Architecture' not supported." }
+    }
+}
+
+function Get-Install-Directory([string]$OS, [string]$Architecture) {
+    $dir = "OpenTelemetry .NET AutoInstrumentation"
+
+    if($OS -eq "windows"){
+        if($Architecture -eq "x86"){
+            return (Join-Path "C:\Program Files (x86)\" $dir)
+        } elseif($Architecture -eq "x64") {
+            return (Join-Path "C:\Program Files\" $dir)
+        }
+    }
+
+    throw "TODO: Not supported yet."
+}
+
+function Get-Temp-Directory([string]$OS) {
+    if($OS -eq "windows") {
+        return $env:temp
+    }
+
+    throw "TODO: Not supported yet."
+}
+
+function Prepare-Install-Directory([string]$InstallDir) {
+    New-Item -ItemType Directory -Force -Path $InstallDir | Out-Null
+}
+
+function Download-OpenTelemetry([string]$OS, [string]$Architecture, [string]$Version, [string]$Path) {
+    $archive = $null
+
+    if($OS -eq "windows") {
+        $archive = "opentelemetry-dotnet-instrumentation-windows.zip"
+    } else {
+        throw "TODO: Not supported yet."
+    }
+
+    $dlUrl = "https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/download/$Version/$archive"
+    $dlPath = Join-Path $Path $archive
+    Invoke-WebRequest -Uri $dlUrl -OutFile $dlPath
+
+    return $dlPath
+}
+
+function Extract-OpenTelemetry([string]$DlPath, [string] $InstallPath) {
+    Expand-Archive $DlPath $InstallPath -Force
+}
+
+function Setup-OpenTelemetry-Environment([string] $OS, [string] $InstallPath) {
+    $target = [System.EnvironmentVariableTarget]::Machine
```

Comment:
I have a strong opinion that we should only set env vars for the current terminal session.

Reply:
These are the most common global envs that hardly change. But I agree that the process env should enable instrumentation (that's why it's not on the list).

Reply:
Btw I would agree with simple ones, but always setting path based env's can easily lead to errors.

Reply:
One more idea is to have a Global switch, so when user is fine with global setup + restart (due machine level variable), then the install script can handle all of the complicated parts.